python/pyspark/sql/streaming/stateful_processor_api_client.py

```diff
@@ -154,6 +154,53 @@ def get_list_state(
             # TODO(SPARK-49233): Classify user facing errors.
             raise PySparkRuntimeError(f"Error initializing value state: " f"{response_message[1]}")
 
+    def is_first_batch(self) -> bool:
+        import pyspark.sql.streaming.StateMessage_pb2 as stateMessage
+
+        is_first_batch = stateMessage.IsFirstBatch()
+        request = stateMessage.UtilsCallCommand(isFirstBatch=is_first_batch)
+        stateful_processor_call = stateMessage.StatefulProcessorCall(utilsCall=request)
+        message = stateMessage.StateRequest(statefulProcessorCall=stateful_processor_call)
+
+        self._send_proto_message(message.SerializeToString())
+        response_message = self._receive_proto_message()
+        status = response_message[0]
+        if status == 0:
+            return True
+        elif status == 1:
+            return False
+        else:
+            # TODO(SPARK-49233): Classify user facing errors.
+            raise PySparkRuntimeError(f"Error getting batch id: " f"{response_message[1]}")
+
+    def get_initial_state_iter(self) -> Tuple[Tuple, "PandasDataFrameLike"]:
+        from pandas import DataFrame
+        import pyspark.sql.streaming.StateMessage_pb2 as stateMessage
+
+        # while True:
+        get_initial_state = stateMessage.GetInitialState()
+        request = stateMessage.UtilsCallCommand(getInitialState=get_initial_state)
+        stateful_processor_call = stateMessage.StatefulProcessorCall(utilsCall=request)
+        message = stateMessage.StateRequest(statefulProcessorCall=stateful_processor_call)
+
+        self._send_proto_message(message.SerializeToString())
+        response_message = self._receive_proto_message()
+        status = response_message[0]
+        if status == 1:
+            return None
+        elif status == 0:
+            iterator = self._read_arrow_state()
+            # we only expect 1 batch from jvm
+            batch = next(iterator)
+            batch_df = batch.to_pandas()
+            # we only expect 1 row at a time from jvm
+            deserialized_key = self.pickleSer.loads(batch_df.at[0, 'key'])
```

Comment:
Correct me if I'm wrong, but do we only get a single row in this call? Would the rest of the initial state records get ignored in this case?